### PR TITLE
[vNext Tokens] Update `CommandBarTokens`

### PR DESCRIPTION
**Pull request by mischreiber**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Controls will start coming fast after this, but I wanted to call out `CommandBar` independently, as it's our first converted UIKit control.

Not much exciting happening here, honestly. The biggest item of note is the addition of an observer for the `didChangeTheme` notification, which allows us to change our theme while visible. This is an enhancement above the existing vNext control, which features no such ability.

It's also worth noting that most of the token protocols assume SwiftUI controls, and thus also expect the structure of our SwiftUI controls. In other words, they want the control to have a `state` object. There's no such object on our UIKit controls, so the control itself will conform to the `ControlConfiguration` protocol.

### Verification

Size improved from 31,768,161 -> 31,735,921, which means an improvement of **~31 kB**. Not bad!

| Default                                      | Custom colors                              |
|----------------------------------------------|--------------------------------------------|
| blue_light | green_light |
| blue_dark | green_dark |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [x] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [x] Different resolutions (1x, 2x, 3x)
- [x] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [x] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/832)

### CODE REVIEW

**Comment by sophialee0416** [comment]
We had a conversation a while back about how the keyboard does not change with the theme because the notification for theme change is sent per window. Will this be changed in the new token system?

**Comment by mischreiber** [comment]
> We had a conversation a while back about how the keyboard does not change with the theme because the notification for theme change is sent per window. Will this be changed in the new token system?

...yes? Let's say yes. Also please file a bug against me when you get a chance so I don't forget to fix that 😬

**Comment by harrieshin** [comment]
how much did you save here? :)

File: ios/FluentUI/Command Bar/CommandBar.swift [hunk 1]
```diff
@@ -39,16 +37,30 @@ open class CommandBar: UIView, FluentUIWindowProvider {
...
 
-        backgroundColor = commandBarTokens.backgroundColor
+        backgroundColor = UIColor(colorSet: tokens.backgroundColor)
         translatesAutoresizingMaskIntoConstraints = false
 
+        self.themeObserver = NotificationCenter.default.addObserver(forName: Notification.Name.didChangeTheme,
+                                                               object: nil,
+                                                               queue: .main) { [weak self] _ in
+            if let strongSelf = self {
+                strongSelf.updateButtonTokens()
```
Comment (harrieshin) [inline ios/FluentUI/Command Bar/CommandBar.swift:47]: so notifcation can be posted to all windows, but updatebuttontokens will figure/filter out if the theme changes for that window is applicable for this instance of the class?

File: ios/FluentUI/Core/ControlHostingContainer.swift [hunk 2]
```diff
@@ -41,6 +41,12 @@ public class ControlHostingContainer: NSObject {
         self.updateRootView()
     }
 
+    deinit {
+        if let themeObserver = self.themeObserver {
+            NotificationCenter.default.removeObserver(themeObserver)
```
Comment (harrieshin) [inline ios/FluentUI/Core/ControlHostingContainer.swift:46]: i thought we don't need to do any removeobserver these days. do we? https://developer.apple.com/documentation/foundation/nsnotificationcenter/1413994-removeobserver

**Review by harrieshin — APPROVED** [review]

File: ios/FluentUI/Core/ControlHostingContainer.swift (hunk 2 above)
Comment (mischreiber) [reply ios/FluentUI/Core/ControlHostingContainer.swift:46]: Unfortunately that's only the case if you use `addObserver(_:selector:name:object:)`. If you observe via `addObserver(forName:object:queue:using:)` then you do still need to deregister once you're done observing.

File: ios/FluentUI/Command Bar/CommandBar.swift (hunk 1 above)
Comment (mischreiber) [reply ios/FluentUI/Command Bar/CommandBar.swift:47]: Good point. At the moment no, this is pretty brute-force logic, and will apply to the entire app. But that's a good optimization: the theme posting could/should include the window, and then we can have logic that decides whether it applies to us or not.

File: ios/FluentUI/Core/ControlHostingContainer.swift (hunk 2 above)
Comment (mischreiber) [reply ios/FluentUI/Core/ControlHostingContainer.swift:46]: That said, I'm adding a little more logic in the notification handler, so I'm switching to `addObserver(_:selector:name:object:)` in both these cases.

**Comment by mischreiber** [comment]
> how much did you save here? :)

31,768,161 -> 31,735,921, so ~31 kB. Not bad!

**Pull request merged**